### PR TITLE
fix: do not generate non-existing paths

### PR DESCRIPTION
**Pull request by P0lip**
Fixes https://github.com/stoplightio/spectral/issues/91

**Checklist**

- [x] Tests added / updated
- [ ] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/rulesets/oas2/__tests__/valid-example.ts [hunk 1]
```diff
@@ -156,7 +156,7 @@ describe('valid-example', () => {
       expect.objectContaining({
         code: 'valid-example',
         message: '"self" property type should be array',
-        path: ['definitions', 'halRoot', '_links', 'self'],
+        path: ['definitions', 'halRoot', 'example', '_links', 'self'],
```
Comment (P0lip) [inline src/rulesets/oas2/__tests__/valid-example.ts]: This is quite tricky. I still need to fix it.

Comment (P0lip) [reply src/rulesets/oas2/__tests__/valid-example.ts]: This is a bit of a different issue, fixed in https://github.com/stoplightio/spectral/pull/459/commits/0e3844f2025cd5fb07d41f217e1c33c5e3d85ef0
yet I believe the new error message we generate is actually worse than the former one, or am I wrong?

File: src/rulesets/oas3/__tests__/valid-example.ts [hunk 2]
```diff
@@ -169,7 +169,7 @@ describe('valid-example', () => {
       expect(results).toEqual([
         expect.objectContaining({
           code: 'valid-example',
-          message: `"ip_address" property format should match format "${format}"`,
+          message: `"example" property format should match format "${format}"`, // hm, ip_address is likely to be more meaningful no?
```
Comment (P0lip) [inline src/rulesets/oas3/__tests__/valid-example.ts]: @philsturgeon thoughts?

Comment (philsturgeon) [reply src/rulesets/oas3/__tests__/valid-example.ts]: Depends? This feels like a limitation of trying to shove everything into a simple string message. We are talking about the example, but it sure could be useful to let them know its the ip_address example that is wrong.

Comment (P0lip) [reply src/rulesets/oas3/__tests__/valid-example.ts]: Yeah, exactly.
I'd expect to receive something like `"ip_address" example property format should match format "${format}"`. Will do it this way.

**Pull request merged**